### ISSUE TITLE
Portrait mode saves each photo in a new directory

### ISSUE THREAD

**Issue by zbraniecki** [nextcloud/android#3271]
### Is your feature request related to a problem? Please describe.
Google Camera on Pixel 2/3 in Portrait mode saves each photo in a new folder.

See:
 - https://forum.xda-developers.com/pixel-2-xl/help/portrait-mode-creates-folder-t3709779
 - https://productforums.google.com/forum/#!category-topic/phone-by-google/qScg2CMrzGA
 - https://productforums.google.com/forum/#!topic/phone-by-google/LyB4OLy528o
 - https://www.reddit.com/r/GooglePixel/comments/7qx6hm/pixel_2s_portrait_mode_makes_folders_for_each/
 - https://androidcommunity.com/gcam-tool-fixes-the-way-modded-google-camera-apps-save-portrait-mode-images-20180220/

This makes camera photo backups impossible because you'd need to add a new folder to backup.

### Describe the solution you'd like

It would be awesome if Nextcloud app could recognize DCIM/IMG_* folders and back their content into Camera upload location together with all other photos.

### Describe alternatives you've considered
I can manually move each portrait to the Camera directory, but that sucks.

**Comment by tobiasKaminsky** [comment]
> 1. Add an option for a camera location to allow recursion, one of:
>    a. Unlimited dirs/subdirs;
>    b. User-defined depth (default to 1); or
>    c. Only one level deep.

For a/b: 
if you have /camera specified and /camera/burstX/img_1-10.jpg are created, what shall happen:
- all files are uploaded into same /remote/camera folder -> could give you file conflicts
- all files are uploaded into /remote/camera/burstX

**Comment by r2evans** [comment]
Is there a reason to deviate from the naming structure that android is already using to deconflict file-naming? (That is, sync the dirs as-is, no renaming.) You don't have to do any translation, and the android naming convention already mitigates name conflicts.

**Comment by doits** [comment]
I think this would be solved by #285 increment 5: *real* one way sync.

In the meantime with the current logic it might be feasible to just scan for every **/*.jpg and upload it as if they were in the parent directory? There could be a setting to enable/disable this, but I think that's it.

If there is a naming conflict said file could be ignored for now (or saved with a suffix), but I don't think there should be any, at least not with the photo app/DCIM folder. So one could restrict this feature for the DCIM folder to be sure for now.

I think syncing camera photos is really important, maybe so important that such quick fix is worth it.

**Comment by jakommo** [comment]
I came here because I saw all those `IMG_*` folders in the nexcloud auto upload view and was thinking that they aren't backed up.
But doing some spot checking on my nexcloud server, it looks like they are actually there.

```
$ find Pixel2XL/ -name "00100dPORTRAIT_00100_BURST20180815175619889_COVER.jpg" -exec sha256sum {} \;
bb2749ebe9afe959041f7bf4a29b54c88883da87297fbb9f2be0514d0d327fbc  Pixel2XL/2018/08/00100dPORTRAIT_00100_BURST20180815175619889_COVER.jpg
bb2749ebe9afe959041f7bf4a29b54c88883da87297fbb9f2be0514d0d327fbc  Pixel2XL/00100dPORTRAIT_00100_BURST20180815175619889_COVER.jpg
$ find Pixel2XL/ -name "00000PORTRAIT_00000_BURST20180815175619889.jpg" -exec sha256sum {} \;
5826105a938bc01519e45aefbc5a5e41886d1951d2f0b3b251b1f3f85f6ff882  Pixel2XL/00000PORTRAIT_00000_BURST20180815175619889.jpg
5826105a938bc01519e45aefbc5a5e41886d1951d2f0b3b251b1f3f85f6ff882  Pixel2XL/2018/08/00000PORTRAIT_00000_BURST20180815175619889.jpg
```

The two files in the top level `Pixel2XL/` folder I have manually uploaded by selecting them from within the corresponding `IMG_` folder and then uploading them.
The ones in the `Year/Month/` folder have been automatically uploaded, but they seem to be the exact same file.
I also searched for those files on my phone and the only place I found them was within the `Camera/IMG_*` folder.

This makes me wonder if there already _is_ something in place that is uploading the Portrait pictures from the `IMG_` sub folders . 

To double check I just took a Portrait picture now and instantly saw the upload notification.
Checking the server, the two files are there 
```
$ find Pixel2XL/ -name "*201909182216*"
Pixel2XL/2019/09/00100sPORTRAIT_00100_BURST20190918221608450_COVER.jpg
Pixel2XL/2019/09/00000PORTRAIT_00000_BURST20190918221608450.jpg
```
and on the phone they only exist within the `IMG_` folder.

**Comment by tobiasKaminsky** [comment]
https://github.com/nextcloud/android/pull/4425

This should solve it, can you give it a try?
There is an APK file in this PR, which you can install in parallel to your existing Nextcloud app.

**Comment by nephthys** [comment]
I also have the problem on my Pixel 4 XL: a new subfolder is created with each photo in Portrait mode.

I tested the latest version of the application Nextcloud on the Play Store and the APK of #4425. It doesn't change anything unfortunately...

The result with the app on the Play Store:
```
IMG_20191102_233616/
IMG_20191102_233616/00000PORTRAIT_00000_BURST20191102233616470.jpg
IMG_20191102_233616/00100trPORTRAIT_00100_BURST20191102233616470_COVER.jpg
```

The result with the APK of #4425:
```
IMG_20191103_003206/
IMG_20191103_003206/IMG_20191103_003206/
IMG_20191103_003206/IMG_20191103_003206/00000PORTRAIT_00000_BURST20191103003206222.jpg
IMG_20191103_003206/IMG_20191103_003206/00100trPORTRAIT_00100_BURST20191103003206222_COVER.jpg
```

Thanks...

**Comment by juliushaertl** [comment]
@tobiasKaminsky I also can still see the issue with portrait photos with the latest playstore release as described by @nephthys 

The Photos get uploaded just fine, the issue is that there is still a new media folder detected for each portrait photo.

**Comment by maddes** [comment]
Tobias I'm having this issue in a Pixel 3a. Will try the APK o your PR.
What's the release plan if that's successfully merged?

**Comment by tobiasKaminsky** [comment]
This seems a veery old bug, which I thought is already fixed.
@juliushaertl can yo confirm that it still a problem for you?

**Comment by tobiasKaminsky** [comment]
So I clarified this.
This is something we cannot /want to fix, as messing/changing around hierarchy while uploading just asks for problems.
So if Pixel is creating new subdirectories, then it is their "fault" and honestly I do not get why they are doing it.

**Comment by juliushaertl** [comment]
Yes, this is still an issue with 3.12.1 RC1

Isn't there a way to somehow ignore directories from showing them in the autoupload if they are a child of another folder that is offered to autoupload?

![Screenshot_20200708-084448](https://user-images.githubusercontent.com/3404133/86886509-5d1f2a00-c0f7-11ea-8e27-465eb4b1643f.png)

As an alternative, how about just showing auto upload folders that are enabled by default and just show the full list after pressing a plus button to setup new auto uploads? That would already help a lot to get a cleaner UI for such cases?

**Comment by AndyScherzinger** [comment]
> Isn't there a way to somehow ignore directories from showing them in the autoupload if they are a child of another folder that is offered to autoupload?

yes that would be possible but isn't implemented atm

> As an alternative, how about just showing auto upload folders that are enabled by default and just show the full list after pressing a plus button to setup new auto uploads? That would already help a lot to get a cleaner UI for such cases?

the simpler solution is probably rather the initial idea of hidding such folders at all times (since they don't make any sense for auto upload)

A question (since i don't have such a device/behavior): Are these images uploaded or ignored at the moment when the parent folder has been activated?

**Comment by juliushaertl** [comment]
> A question (since i don't have such a device/behavior): Are these images uploaded or ignored at the moment when the parent folder has been activated?

They are uploaded just fine with the existing auto upload of the parent folder and the subdirectory is also properly kept:

![image](https://user-images.githubusercontent.com/3404133/86896564-695eb380-c106-11ea-82ee-63e39732bd89.png)

**Comment by maddes** [comment]
Yes, they are uploaded just fine, the only problems I see with it are:

1. A small annoyance of all those folders showing up on the auto-upload list of possible folders to auto-upload.
2. A bigger annoyance: it seems to create broken images on the Google Photos gallery. Looks like the app creates/saves some thumbnails that Photos can't read? Basically since I enabled the auto-upload I see a bunch of broken (gray) images on the Google Photos library after every sync.

![image](https://user-images.githubusercontent.com/1455722/86949860-af773f80-c125-11ea-87f7-3e297f08f74e.png)

**Comment by maddes** [comment]
One more snap showing the details of one of those:

![image](https://user-images.githubusercontent.com/1455722/86950052-ff560680-c125-11ea-80ed-74d0f30a2c12.png)

**Comment by erikkroes** [comment]
I also get a lot of "New photo media folder detected" and "File upload conflict" with the Pixel 3a. It has a negative effect on the WAF (Wife Acceptance Factor) in this house and my general mood.

**Comment by tygill** [comment]
I want to chime in and add another request for hiding subfolders of folders which are already selected for auto upload.

One particular annoyance we've had with the 'New photo media folder detected' notifications is that there isn't a wya to hide them without hiding all NextCloud notifications. Since my wife alternates between taking pictures and videos of our kids often, lots of her videos get vibrations recorded because the previously taken pictures trigger the notifications.

Ultimately, since the pictures are already being uploaded (because the parent folder is uploading them), the notifications feel pretty unnecessary and redundant. And unfortunately NextCloud doesn't seem to have granular control over which notifications to show either.

**Comment by doits** [comment]
@tygill See #4738 for hiding the new folder notification - it is possible to do it without hiding all notifcations! You should be able to expand the notification and do it there.

**Comment by tygill** [comment]
Interesting - it looks like #4738 does allow this notification to be hidden, which is exactly what I wanted. That's excellent to know. Thanks for pointing it out!

**Comment by joshtrichards** [comment]
This seems to be addressed by #4425 & #4738
